### ISSUE TITLE
need to add maven dependency to include tools.jar in ate

### ISSUE THREAD

**Issue by peidong-hu** [bigtester/automation-test-engine#64]
jdk javac in tools.jar is used to dynamically compile caserunner. but the user might only install jre which has tools.jar missing.

ate needs to find a way to install tools.jar in it's maven dependency and change caserunnergenerator code to accommodate this.

**Comment by Jun-Yang** [comment]
Maybe we can add tools dependency in POM:

<profiles>
    <profile>
      <id>default-tools.jar</id>
      <activation>
        <property>
          <name>java.vendor</name>
          <value>Sun Microsystems Inc.</value>
        </property>
      </activation>
      <dependencies>
        <dependency>
          <groupId>com.sun</groupId>
          <artifactId>tools</artifactId>
          <version>1.4.2</version>
          <scope>system</scope>
          <systemPath>${java.home}/../lib/tools.jar</systemPath>
        </dependency>
      </dependencies>
    </profile>
  </profiles>


2015-02-14 2:24 GMT-04:00 peidong-hu <notifications@github.com>:

**Comment by peidong-hu** [comment]
Yes. I have used the similar approach for another lib before. It should
work. But not sure if we need to or how to change caserunnergenerator yet
to work with it. (Some article says that code needs to be changed if use
maven to include the lib.)

**Comment by peidong-hu** [comment]
a workaround is to check if the system has jdk installed or not, and give user proper message to install jdk if only jre found.
